connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/RestClient.java

```diff
@@ -129,6 +130,7 @@ static <T> HttpResponse<T> httpRequest(HttpClient client, String url, String met
 
             if (sessionKey != null && requestSignatureAlgorithm != null) {
                 InternalRequestSignature.addToRequest(
+                    CryptoLibrary.SYSTEM,
```

Comment:
Food for thought: might be worth adding the `CryptoLibrary` instance as a constructor parameter once https://github.com/apache/kafka/pull/12828 is merged.